MLH-Fellowship/Social-BERTerfly
# Bert model

### Changes made:

Should solve #2 and #3 .

- Setup a basic Flask backend
- Setup endpoint `/predict` for fetching predictions
- Setup Fetch API request and response handling to the model
- Create a `predict_type` function in `predict_types.py` to return the predicted user personality type.
- Return a json response from the endpoint to display the response in following format:
` {"type" : <personality_name> }`

### Tests to be checked
- Check by entering some text in the browser form by running `index.html` on your browser and `flask run` on your terminal together.
- Check browser console for the expected json response.
- Check your terminal to see if the whole BERT model gets downloaded or no. It will probably download the very first time you run it, (warning, it's 500 MB). 
- **Make sure BERT model doesn't download for all but the first time that you test.**
- For the next few times, keep track of how long it takes to print the response on the browser console after sending the request. For me, it's around 1 min (which we can account for by keeping a loader, or nothing even)

### What to do next:
- Test same on Twitter API responses.
- Return probabilities of each class instead of one label. 
- Send this response json to frontend for displaying on the page.
